src/opendr/perception/object_detection_2d/centernet/centernet_learner.py

```diff
@@ -395,16 +395,9 @@ def infer(self, img, threshold=0.2, keep_size=True):
 
         self._model.set_nms(nms_thresh=0.45, nms_topk=400)
 
-        if isinstance(img, Image):
-            _img = img.numpy()
-        elif isinstance(img, np.ndarray):
-            _img = img
-        else:
-            if img is not None:
-                error_msg = "Input should be of type Image or numpy array."
-            else:
-                error_msg = "Input image is of type None."
-            raise ValueError(error_msg)
+        if not isinstance(img, Image):
+            img = Image(img)
```

Comment:
This doesn't exactly reproduce the previous code.

Reply:
Code added to ensure that valid types are passed into Image constructor.